### DISCUSSION TITLE
CP-SAT Implement breaks with cumulative constraint

### DISCUSSION THREAD

**Discussion by ajouzali** [google/or-tools#2808] [CP-SAT questions]
We have a complex job-shop problem in which we have a lot of machine breaks with different sizes. With the large number of tasks, the spliting of the tasks using optional interval cause a very big number of variables in our CP-SAT model. So, we are trying to find an alternative implementation. So, the idea is as follows: 
- to create a second time dimension X which only increases in the usefull production time and 0 when there is a break

- So, the constraint should be at the end as follows
1- start(Task) and end(Task) in usefull production interval
2-  X(end(Task)) - X(start(Task)) == Duration(Task)

The problem is our intervals and durations are in seconds and the mapping  form and to X would also cause a lot of variables. 
Is there any recommendation on how to deal with this case? reference, or an Example?
![concept](https://user-images.githubusercontent.com/91462620/135816597-9c75f4b7-0a0d-4a9d-9e1e-68fafb60b200.png)

**Comment by lperron** [comment]
Hi, have you looked at https://github.com/google/or-tools/blob/stable/ortools/sat/doc/scheduling.md#intervals-spanning-over-breaks-in-the-calendar ?

**Reply by ajouzali** [reply]
We tried a similar implementation with multi-spanning intervals but we are not happy with the performance. So, we are thinking about a similar concept to what is implemented in routing package (cumulative dimension with map evaluator ). But the seconds grid we use does not help also.

**Comment by gregy4** [comment]
Can you give us some example ?  It is not very helpful to talk generally about large number of tasks or that you are not happy with the performance.

**Comment by Hakunari** [comment]
Hello! Did you get any solution? I got the same problem.
